src/components/typography/list.component.tsx

```diff
@@ -1,15 +1,21 @@
 import React from "react";
 import propTypes from "prop-types";
-import Typography from "./typography.component";
+import { SpaceProps } from "styled-system";
+import Typography, { TypographyProps } from "./typography.component";
 
-const getListStyleType = (as) => {
+export interface ListProps extends SpaceProps, TypographyProps {
+  as?: keyof JSX.IntrinsicElements;
```

Comment:
```suggestion
  as?: "ul" | "ol";
```

Reply:
It's also worth noting that `as?: keyof JSX.IntrinsicElements` is already in `TypographyProps`, so if we really did want to allow any such element name to be supplied then there would be no point adding this to `ListProps`

Reply:
@mkrds Yeah this is fair - I was battling Typescript giving non-descript error messages realised a semi solution but yeah it would make sense to just be more specific for sure.

@robinzigmond As I recall I couldn't remove the linting typescript errors unless I defined the "as" again for ListProps. So yeah I thought that would be the case too. Perhaps I'm wrong though

Reply:
Setting as to be specific elements like this causes conflicts with the as prop in general. Also, following feedback from Jamie we shouldn't be so restrictive to limit it to specific strings as a user should be able to pass in their own web component. So with that said the type for this is now 'React.ElementType'.